### DISCUSSION TITLE
How do you calculate the amount of energy leaving the observable region?

### DISCUSSION THREAD

**Discussion by mikehelland** [a-cosmology-group/acg#184] [Astrophysics and Physics]
_**Note: Hopefully this problem and its solution is generic enough to be useful to a variety of models, not just one.**_
_Update: the problem has been restated to be as clear as possible, and I've included updates on my attempt_
_Update Nov26: I think I have some results, reported below_

----

### The problem, cosmologically speaking

Assume a volume that has an average of N light sources per cubic unit, each with an average luminosity of L. Within this volume is a sphere with radius R. 

Assume that each light source, for whatever reason, reaches a maximum radius of R, having a power factor (?) of $1 - r/R$ from its source. 

_Note: this loss is in addition to the inverse square law._
_Note2: I changed this to_ $(1 - r/R)^2$ _so there's a loss due to redshift, and a loss due to time dilation._

How much energy from sources inside the sphere gets out? How much energy from outside the sphere gets in? Presumably, they should be equal. Can the energy per second coming in be related to a temperature T? Thus creating a relationship between R, N, L and T?

----

### The problem, mathematically speaking

I decided to start 2D. I made a main circle $C_0$. I started at the perimeter, lined it with boxes, and worked my way in. Obviously, zero energy is getting out here. This just fills the circle. So now choose any box in the circle, and do the same thing for it, make a bunch of boxes. Some of this will go outside the original circle. 

https://mikehelland.github.io/hubbles-law/other/cmbenergy.htm

I then extended it to 3D:

![](https://raw.githubusercontent.com/mikehelland/hubbles-law/master/img/cmbenergy3d.png)

(just one moveable sphere)  
https://mikehelland.github.io/hubbles-law/other/cmbenergy3d.htm

(does all possible spheres)  
https://mikehelland.github.io/hubbles-law/other/cmbenergy3dall.htm

Using this page I bumped the left slider down to "10". That's numerically integrating the sphere with a dx of 10. R in the model is 140. So if the units are in hundred million light years, 10 is 1 billion light years. 

----

### Results

I think my conclusion is that, assuming N = 1 light source per cubic billion light years, each has a luminosity of L = 1 watt, and the radius is R = 14 billion light years, then:

* 15 megawatts is landing outside the main sphere

I'm not sure what this really tells me. I think what I want to know is the luminosity of the light that intersects the main sphere's surface area. That's:

* 0.2 megawatts

So I guess you divide that by the surface area? And then you increase N and L until you get... I was trying to get to a temperature somehow. Now I've kind of lost track of how I was going to do that. 

### Temperature

Assuming we can use:

$$\displaystyle T={\sqrt[{4}]{\frac {L}{4\pi R^{2}\sigma }}}$$

    const pc2ly = 3.261564
    const pc2km = 3.08e19

    var L // Luminosity of a box
    var Lsol = 3.846e26
    var suns = 2e11 // 200 billion suns in a galaxy

    var R = 140

    // in meters
    var Rm = R/10 / pc2ly * pc2km * 1000 * 1000

    const sbc = 5.67e-8 //Stefan-Boltzman constant

    T = (L * Lsol * suns / (4 * pi * sbc * Rm**2))**(1/4)

I get a temp of 0.188 K.

That's assuming 1 galaxy per cubic giga-lightyears, with the luminosity of 200 billion suns.

To find the right N (number of galaxies per Gly^3) and average luminosity of a galaxy L, it seems I'm only off by a factor of 50K somewhere:

    (50000 * L * Lsol * suns / (4 * pi * sbc * Rm**2))**(1/4) 

2.8 K

----

Any comments or suggestions?

**Comment by HanDeBruijn** [comment]
>Let's say there's a spherical volume of space with a radius of R, with galaxies uniformly distributed within.

Can we call that our Observable universe, with a Cosmological horizon at R ?

>Assume that to an observer at the center of the volume, light coming from R has a z of infinity. So, light leaving the center has a z of infinity at R.

Does that mean that there is no preferred (space-time) position for an observer ?
However, I don't think there is a _z of infinity_; instead a maximal redshift $z=e^W-1\approx 28.5$ may be theoretized

>Based on these assumptions, no light from the center will make it outside of the volume.

Not exactly. There is an interesting publication, called _[Relativistic Escape Velocity using Special Relativity](http://www.mrelativity.net/MBriefs/Relativistic%20Escape%20Velocity%20using%20Special%20Relativity.htm)_. See [thread 153](https://github.com/orgs/a-cosmology-group/discussions/153#discussioncomment-7123672).
Please follow the reasoning in that paper up to Eq. (7). Then I propose to continue as follows.

$$
m_0c^2 - \frac{G M m_0}{R} = m_0c^2\sqrt{1-\frac{v^2}{c^2}}
$$

There is a _removable singularity_ for $v=c$ whch is purely mathematical and defnitely _not physical_.
This makes our Special Relativity analysis essentially different from its GR counterpart and much less problematic.
But _using a computer mathematical application_ as suggested in the paper is not necessary:

$$
\left(1 - \frac{G M}{R c^2}\right)^2 = 1-\frac{v^2}{c^2}
$$

$$
v = c\sqrt{1 - \left(1 - \frac{G M}{R c^2}\right)^2} = \sqrt{\frac{2GM}{R}-\left(\frac{GM}{Rc}\right)^2}
$$

So the escape velocity according to SR is

$$
v=\sqrt{\frac{2GM'}{R}} \quad \mbox{with} \quad M' = M + \left[-\frac{GM^2}{2Rc^2}\right]
$$

The term between square brackets is recognized as the negative mass of the [Self Energy Field](https://www.grc.nasa.gov/www/k-12/Numbers/Math/Mathematical_Thinking/possible_scalar_terms.htm) of a body with radius R and mass M.
Let's specialize this for our model of the Observable universe, the one I've been trying to explain in [thread 179](https://github.com/orgs/a-cosmology-group/discussions/179#discussioncomment-7474808).

$$
M'_O = \frac{W}{2}\frac{c^3}{G.H}(1-g) \quad ; \quad g = \frac{W^2}{4}\mbox{VMT} \quad ; \quad R_O = W\frac{c}{H}
$$

$$
v = \sqrt{2G\frac{W}{2}\frac{c^3}{G.H}\frac{H}{Wc}(1-g)} \quad \Longrightarrow \quad \Large\boxed{v = c\sqrt{1-g}} \normalsize \approx 0.9671398994 \times c
$$

With Special Relativity instead of GR, the **escape velocity** for our Observable ("black hole") universe is calculated to be **slightly less than the speed of light**. As a consequence, light **can** escape from - as well as enter into - that part of the universe. Contradictory to
>Based on these assumptions, no light from the center will make it outside of the volume.

**Reply by mikehelland** [reply]
Thanks for the response.

I get that you may disagree with the assumptions, but that's the problem I'm interested in. 

I'll update the original post with the steps I've broken it into so far.

**Reply by RedshiftDrift** [reply]
@mikehelland 
What are your assumptions?
- Which function $z(d)$ gives the redshift vs distance $d$ from the source?  (We know $z(0) = 0$ and $z(R) = \infty$, but what's the function in between?)

Since the function $f(d)$ is probably not easily integrable in spherical coordinates, why not just integrate numerically?

- Are there galaxies outside the radius R? (*"the amount of energy leaving the volume... should tell us the amount of energy entering the volume"* seems to indicate that you assume a uniform distribution of galaxies everywhere, correct?)

If the distribution of galaxies is uniform everywhere, then the power leaving the spherical shell of radius $R$ is the same as the power that enters the sphere: the net power transfer is zero.

P.S. "Watts" is **power = energy-per-unit-time**, but the question is about energy; did you want to say "power leaving the observable region"? (You can edit the question b.t.w.)

**Reply by mikehelland** [reply]
> Which function gives the redshift vs distance from the source? (We know and , but what's the function in between?)

r = z/(1+z)

But I recommend we don't even use z. z is prejudiced toward increasing wavelengths.

$$1+z = \frac{\lambda_{obs}}{\lambda_{emit}}$$

When wavelength increases, z is positive. When z goes to infinity, so does wavelength.

When wavelength decreases, z is negative. When z goes to -1, wavelength goes to 0.

Redshift has an infinite range, and blueshift has a finite range (0 to -1).

We should be concerned with a reduction of frequency and/or energy. For that, let's quantify blueshift as a positive number instead, so:

$$1+b = \frac{1}{1+z} = \frac{f_{obs}}{f_{emit}}$$

Now redshifts are when b is negative, and they can only go to negative -1. The range of redshifts is now finite.

Now r = -b. So:

$$1-r = \frac{1}{1+z} = \frac{f_{obs}}{f_{emit}}$$

$$b = \frac{1}{1+z} - 1 = -\frac{z}{1+z}$$

**Reply by mikehelland** [reply]
> the net power transfer is zero.

That's what I'm counting on though. The amount of energy leaving (joules per second , so watts, right?) should be equal the amount coming in.

But the amount coming in will "die" out before it reaches the center. So there's a specific amount of energy entering our observable region, but never making it to us. The idea is to connect that to a temperature somehow. If that temperature is set to 2.7 K, then that should be able to determine R.

**Reply by bgreenman** [reply]
I Think that HanDiBrujin's response to your question is correct

**Comment by HanDeBruijn** [comment]
>Let's say there's a spherical volume of space with a radius of R, with galaxies uniformly distributed within.

Isn't that the same as a uniform mass density distribution $\rho_0$ ? No need to do it numerically; you can do it analytically.
I think that a handful of useful ingredients has been provided in [thread 127](https://github.com/orgs/a-cosmology-group/discussions/127).
(I'm leaving out the Variable Mass Theory in the sequel because nobody seems to be understanding or interested)
If $M_O$ is the mass and $R_O$ is the radius of your (Observable) universe then we have

$$
M_O = \frac{4}{3}\pi R_O^3\rho_0
$$

There are two ways to define the maximal radius of a black hole (universe), the common General Relativistic (**E**instein) one and the one [derived](https://github.com/orgs/a-cosmology-group/discussions/184#discussioncomment-7565112) with Special Relativity and **N**ewtonian Dynamics.

$$
R_E = \frac{2GM_O}{c^2} \quad \mbox{or} \quad R_N = \frac{GM_O}{c^2}
$$

And so

$$
R_O = G\frac{(8 \mbox{ or } 4)}{3 c^2}\pi R_O^3\rho_0 \quad \Longrightarrow \quad R_O^2 = \frac{3 c^2}{(8 \mbox{ or } 4)\pi G\rho_0}
$$

Repeated for convenience, from that previous thread as mentioned:

$$
H_0 = \sqrt{\chi G \rho_0} \quad \mbox{with} \quad \chi = 8\pi \times
\begin{cases}
1/8 & (: \mbox{OI}) \\
1/6 & (: \mbox{Mach}) \\
1/3 & (: \Lambda\mbox{CDM})
\end{cases}
$$

Combine this with the formula for $R_O$ :

$$
G\rho_0 = \frac{H_0^2}{\chi} \quad \Longrightarrow \quad R_O^2 = \frac{c^2}{H_0^2} \frac{3\chi}{(8 \mbox{ or } 4)\pi}
$$

Your radius of the Observable universe is thus proportional to the Hubble length $c/H_0 \approx 13.3 \mbox{ } ly$.

$$
R_O = \sqrt{\frac{3\chi}{(8 \mbox{ or } 4)\pi}} \frac{c}{H_0} =\sqrt{\mbox{mh}}.\frac{c}{H_0}
$$

where you can choose from

$$
\mbox{mh} = \left(\frac{3}{8} \vee\frac{1}{2}\vee\frac{3}{4}\vee 1\vee 2\right)
$$

It follows anyway that there is a maximum ${\bf R_O \le } \frac{c}{H_0}\sqrt{2} \approx {\bf 18.8 \mbox{ } ly}$ .

**Reply by mikehelland** [reply]
@HanDeBruijn 

>Isn't that the same as a uniform mass density distribution ? 

I don't think so, but it's a good question and might help explain what I'm doing.

If we say there are N galaxies per 1 cubic Gigaparsec in a volume V, and their average luminosity is L, 

If we can find how might light is hitting the surface area of that volume from the inside, that should tell us how much light is hitting the surface area of that volume from the outside.

So, flux might be what the problem is calling for. 

In any case, then we should have a relation to V, N, L, and hopefully from the flux, some kind of temperature.

V, N, L, and the temperature can all be measured. The temperature is 2.7 K.

V should have a radius of about 14 billion light years. 

What's N and L?

**Reply by HanDeBruijn** [reply]
This formula of yours is found in [thread 160](https://github.com/orgs/a-cosmology-group/discussions/160#discussioncomment-7153006).

$$
d_T = \frac{c}{H_0} \log (1+z) \quad \Longrightarrow \quad 1+z = e^{H_0/c.d_T}
$$

When combined with my previous posting, does it mean that

$$
1+z = e^{H_0/c.R_0} = e^{H_0/c.\sqrt{\mbox{mh}}.c/H_0}  = e^{\sqrt{\mbox{mh}}} \quad \mbox{ ?? }
$$

Because in that case we have a **limited range** of maximal available redshifts:

$$
z = e^{\sqrt{3/8\ \vee\ 1/2 \ \vee\ 3/4 \ \vee\ 1 \ \vee\ 2}} - 1 \lt 3.12
$$

for a galaxy nearby the horizon of your Observable universe. This outcome is contradictory to [observations](https://github.com/orgs/a-cosmology-group/discussions/183).
I guess you don't want to give up your [Time Dilated Past hypothesis](https://github.com/orgs/a-cosmology-group/discussions/160) ?

**Reply by bgreenman** [reply]
HanDeBrujin is correct

**Reply by mikehelland** [reply]
@HanDeBruijn 

In your second equation you equate $d_T$ to $R_0$.

That's incorrect.  $d_T$ goes from 0 to infinity, it is $-ct$ where $t$ has time dilation. $t = -14$ billion years. Take time dilation out of -14 billion years, you get:

$$\tau = \frac{1}{H} (e^{Ht} - 1) = \frac{1}{H} (e^{-1} - 1) = -8.85~ \text{billion years}$$

And this equates to $z=1.718$, so it's not any type of special time or place.

(Assuming $1/H = 14$ billion years.)

**Reply by bgreenman** [reply]
Its not easy for a novice to calculate the amount of energy leaving the observable region ofthe universe

**Reply by mikehelland** [reply]
What answer did you get and how did you get it?

**Reply by HanDeBruijn** [reply]
@mikehelland .
>That's incorrect.

How so? That time $t$ is your Measured **t**ime, right? 
Strictly speaking _that_ is incorrect. What is really _Measured_ - or better: estimated - are _distances_, via the [Cosmic distance ladder](https://en.wikipedia.org/wiki/Cosmic_distance_ladder). So-called [Cosmic time](https://en.wikipedia.org/wiki/Cosmic_time) on the other hand is very much (**Big Bang**) model dependent (so I don't trust it). In an eternal and infinite universe - your Observable universe must be a part of it - Measured time is calculated from Measured distance, simply by $-t = r/c$ where $r=$ distance and $c=$ speed of light.
I cannot imagine that time **un**dilated past $\tau$ is relevant in this context :-(Perhaps that's a shortcoming of my imagination).

**Reply by HanDeBruijn** [reply]
@mikehelland . Maybe interesting for you too [in thread 182](https://github.com/orgs/a-cosmology-group/discussions/182#discussioncomment-7606676) one of the CMB pioneering papers.

**Reply by mikehelland** [reply]
@HanDeBruijn 

Not quite.

Imagine a universe where distances are static and time is invariant. $r = -ct$. Everything easy there.

Except, when we observe a supernova for example, we observe it to be time dilated. If we say the time between peak brightness and when its dimming rate drops is 25 days, that includes the effects of time dilation. The time we measure is stretched. That's why I call $t$ "measured" time.

So suppose a universe with static distance, but time is stretched, depending on how far away it is and/or how long ago it happened. Now $-ct > r$. Because time stretched, but distance didn't.

If we take the time dilation out of $t$ we get Minkowski time $\tau$.

So when $-ct = R$, big R for radius of the observable region, that's just when the time dilated time is -14 billion years, which corresponds to an untime dilated time of -8.8 billion years, therefore, a $r$ of 8.8 billion light years and a $z$ of 1.7. Nothing remarkable about that time or place.

The horizon is when $-c\tau = r = R$ and $z$ is infinity.

**Reply by HanDeBruijn** [reply]
@mikehelland .

>The horizon is when $-c\tau = r = R$

Yes of course. Within the framework as currently presented little else is possible. Let's elaborate a bit.

$$
1-H_0\tau = 1+H_0.r/c = 1+z = e^{-H_0.t} \quad \Longrightarrow \quad z = H_0/c.R
$$

which is the _very_ classical (linear) law as originally formulated by Hubble.
Independent of this, the rest of my musings are still going strong, if I'm not wrong:

$$
R = R_O = \sqrt{(mh)}.\frac{c}{H_0} \quad \Longrightarrow \quad z = \sqrt{\frac{3}{8} \vee\frac{1}{2}\vee\frac{3}{4}\vee 1\vee 2} \le 1.414
$$

a value for $z$ which is far away from
>$r = R$ and $z$ is infinity.

And how can $z$ be infinite while $R$ is finite in $z=H_0/c.R$ ?

**Reply by mikehelland** [reply]
I'm not sure what $\sqrt{mh}$ is, or where your last formula comes from.

In my model:

$$r = \frac{c}{H} \frac{z}{1+z} $$

But my model also has nothing to do with the topic of this thread.

**Reply by HanDeBruijn** [reply]
@mikehelland .

The numerical variable $mh$ is connected with the different cosmological models that I [have considered](https://github.com/orgs/a-cosmology-group/discussions/184#discussioncomment-7599186). But you can take $\mbox{mh}=1$ as the value that that suits you best - I think so - giving $R=c/H \approx 13.3 \mbox{ } ly$ .
>In my model:

$$
r(z) = \frac{c}{H} \frac{z}{1+z}
$$

Then  indeed you get as the minimum and maximum for $r(z)$ :

$$
r(0) = 0 \quad \mbox{and} \quad r(\infty) = \frac{c}{H} = R
$$

Ingenious. But quite recently you have told us a different story, resulting in

$$
r=\frac{c}{H}z \quad \Longrightarrow \quad r(0) = 0 \quad \mbox{and} \quad r(\infty) = \infty
$$

It follows from what you have said with $-c\tau=r$.
>But my model also has nothing to do with the topic of this thread.

Oops ???

In response to @bgreenman the same remark is uttered and:

>The redshift-distance modulus made by the Pantheon+SH0ES is exquisite for any purposes.

Do you mean this one?

$$
d_L = \frac{c}{H_0} (1+z) \log (1+z) \quad \Longrightarrow \quad r = d_L \approx \frac{c}{H} (1+z) z \ne \frac{c}{H} \frac{z}{1+z}
$$

One cannot juggle with formulas at will.

**Reply by RedshiftDrift** [reply]
Just a note @HanDeBruijn, from a "curve fitting to SNe data" point of view the logarithmic distance is better fitted with a square-root in the function, as in

$$
d_L = \frac{c}{H_0} (1+z) \log (1+z) \quad \Longrightarrow \quad r = d_L \simeq \frac{c}{H_0} \sqrt{1+z} \  z
$$

for redshft $z \lesssim 7$.

@mikehelland the data fails to fit the very different function $\frac{c}{H_0} \frac{z}{1+z}$ already for $z \ll 1$, so that can't be right.

**Reply by mikehelland** [reply]
Redshift drift, you are correct, that's not the fit for the function.

None of the replies (except your original one) seem to address the topic of this discussion. Han keeps confusing different aspects of my model, randomly, out of context. Here's an accurate summary (though it doesn't really belong in this topic):

* $H$ - constant with dimension inverse time

* $t$ - Measured time, has time dilation

$$-\infty < t \leq 0$$

* $\tau$ - Minkowski time, time dilation removed

$$-H^{-1} \leq \tau \leq 0$$

$$\tau = H^{-1} (e^{Ht} - 1)$$

* $r$ - Physical distance

$$r = -c\tau = \frac{c}{H} \frac{z}{1+z}$$

* $-ct$ - the speed of light x measured time. This value isn't particularly physically meaningful. In standard cosmology, it would be the "light travel time distance." In my model, it's just the "time-dilated light travel time distance" which doesn't actually tell us much about anything, but is used to find a luminosity distance.

$$-ct = -\frac{c}{H} \log (H\tau+1) = \frac{c}{H} \log (1+z)$$

* $d_L$ - luminosity distance. This is a hypothetical distance that essentially says, imagine a universe without expansion or time dilation, or tired light or variable mass. How far away would something be in a plain vanilla spacetime, to look like it does in our spacetime? 

$$d_L = -ct(1+z) = \frac{c}{H} (1+z) \log (1+z)$$ 

That's a decent fit for the SNe data, at H=70.4 km/s/Mpc.

**Reply by HanDeBruijn** [reply]
>Here's an accurate summary
>- $t$ - measured time

One might think that _measured_ distance then follows as $r=-ct$ . But no:
>- $\tau$ - Minkowski time

By sheer logic, it must be then that $r=-c\tau$ and so $r=c/H.z$ . But no:
>- $r$ - Physical distance

$$
r = -c\tau = \frac{c}{H} \frac{z}{1+z}
$$

>Han keeps confusing different aspects of my model

Yes, I'm completely at lost. And wish you the best. I give up.

**Reply by mikehelland** [reply]
"By sheer logic" involves some assumptions that I'm not making, apparently.

**Reply by mikehelland** [reply]
@HanDeBruijn , I've thought about what you said, and I think it's just a case of "measured time" being a bad name, and unintended implications of it.

In my model (which again, isn't relevant here so maybe we can discuss this in another thread) time is dilated, and distances are static. t, x, y, z.

When you take the time dilation out of time, you get a new system of coordinates, $\tau$, x, y, z, but only time has changed. These coordinates match Minkowski spacetime. 

To differentiate the two times, I used Minkowski time, which seems like a good name, and "measured" time as the other one. 

The space coordinates are the same in both systems. Calling $-ct$ "measured distance" makes sense, name-wise. It should be made clear though that the x, y, and z coordinates in the system with measured time are _not_ affected by time dilation, and are not stretching with time. Otherwise you'd be describing a metric with a scale factor, ie, an expanding universe.

In any case, only Euclidean coordinates are required for the problem I've  stated in this topic.

**Reply by HanDeBruijn** [reply]
![fout2](https://github.com/a-cosmology-group/acg/assets/135433222/ee1736e0-0bd3-4a00-81f0-5016d7aeb706)

**Reply by HanDeBruijn** [reply]
@mikehelland . Can't resist ..

Maybe a proper forum for presenting your problem (without any cosmological details please) is [Mathematics Stack Exchange](https://math.stackexchange.com/). I've found a few Questions there that seem to be resemblant to yours:

- [Expected overlap of n circles of equal area randomly placed inside a circle of larger area](https://math.stackexchange.com/questions/3291226/expected-overlap-of-n-circles-of-equal-area-randomly-placed-inside-a-circle-of-l)
- [How find integral of circle or sphere with another circle(sphere) inside?](https://math.stackexchange.com/questions/4639585/how-find-integral-of-circle-or-sphere-with-another-circlesphere-inside)
- [Finding area inside circle and outside another circle](https://math.stackexchange.com/questions/1748346/finding-area-inside-circle-and-outside-another-circle)

**Reply by RedshiftDrift** [reply]
@mikehelland is $R = \frac{c}{H}$?  In any case, the numerical integration seems the simplest approach.

**Comment by bgreenman** [comment]
More work is needed on measuring distances, HanDeBrujin is right

**Reply by mikehelland** [reply]
The redshift-distance modulus made by the Pantheon+SH0ES is exquisite for any purposes.

https://github.com/PantheonPlusSH0ES/DataRelease/tree/main/Pantheon%2B_Data/1_DATA 

Except I don't see how that's at all relevant to the questions posed by this discussion topic.

**Reply by bgreenman** [reply]
Interesting

**Comment by HanDeBruijn** [comment]
@mikehelland .
Maybe it’s better practice to leave the original text of your question intact and augment it with a clearly marked **Update**. Because now people’s quotes and accompanying responses have become vacuous.

**Reply by bgreenman** [reply]
Agree

**Comment by HanDeBruijn** [comment]
@mikehelland . Your last approaches seem to be be similar to those in comments at [Cosmology Slides: asking for reviews](https://github.com/orgs/a-cosmology-group/discussions/189#discussioncomment-7663302).

**Reply by bgreenman** [reply]
I don't know

**Reply by mikehelland** [reply]
@HanDeBruijn 

Similar. 

Mine is pretty basic, and I found a way to simplify it and I think get more accurate results.

    const mly2mpc = 1 / 3.261564
    const mpc2km = 3.08e19
    const gly2m = 1000 * mly2mpc * mpc2km * 1000

    var Lsol = 3.846e26
    var suns = 2e11 // 200 billion suns in a galaxy

    // these three values should produce a temperature T
    var R = 14
    var N = 50000
    var L = Lsol * suns
    
    //Stefan-Boltzman constant
    const sbc = 5.67e-8 

    // for each box
    box.flux = (N * L * (1 - r/R) ** 2) / (4 * pi * (r * gly2m) **2)

Basically around N=50000 gives the CMB temp so 50 thousand Milky Way-ish sources per Gly^3 seems to work out.

**Reply by bgreenman** [reply]
HanDeBrujin answer is correct
